pkg/workflows/wasm/v2/runner_wasip1.go

```diff
@@ -17,14 +16,8 @@ func versionV2()
 //go:wasmimport env switch_modes
 func switchModes(mode int32)
 
-func NewDonRunner() sdk.DonRunner {
-	switchModes((int32)(sdkpb.Mode_DON))
-	return newDonRunner(runnerInternalsImpl{}, runtimeInternalsImpl{})
-}
-
-func NewNodeRunner() sdk.NodeRunner {
-	switchModes((int32)(sdkpb.Mode_Node))
-	return newNodeRunner(runnerInternalsImpl{}, runtimeInternalsImpl{})
+func NewRunner[T any](parse func(configBytes []byte) (T, error)) sdk.Runner[T] {
+	return newRunner[T](parse, runnerInternalsImpl{}, runtimeInternalsImpl{})
```

Comment:
nit/ could drop the name from the signature and just have the param say more, and would `C` be more intuitive for a config type than `T`? :shrug: 
```suggestion
func NewRunner[C any](parseConfig func([]byte) (C, error)) sdk.Runner[C] {
	return newRunner[C](parseConfig, runnerInternalsImpl{}, runtimeInternalsImpl{})
```
We could even use a decorative type (alias?) if we wanted to be super readable, but maybe that's unnecessary:
```suggestion
// Config is....
type Config any

func NewRunner[C Config](parseConfig func([]byte) (C, error)) sdk.Runner[C] {
	return newRunner[C](parseConfig, runnerInternalsImpl{}, runtimeInternalsImpl{})
```

Reply:
Will for the first part. I realized I have more changes we discussed that I didn't add yet. I'll do that in the next commit with those changes.